### PR TITLE
Add UX for importing ballot configurations automatically in BSD

### PR DESCRIPTION
**Pull request by carolinemodic**
Adds the workflow and UI to import ballot configurations that were automatically exported from EM into BSD. I wasn't entirely sure what the best UI would be for the initial "no usb" screen or the "no ballot packages found" screen so definitely open to feedback/suggestions on those. 

When ballot packages are found and one is selected to configure (everything starting with the uploading ballot 1 screen and following is unchanged from before)
Initial Screen when there is no mounted USB: 

Screen Shot 2020-12-09 at 10 55 27 AM

Screen when there is a USB but no valid ballot packages are found: 

Screen Shot 2020-12-09 at 10 55 55 AM

Screen when there is a USB with ballot packages to select from: 
Screen Shot 2020-12-09 at 11 39 59 AM

Note that the first two commits are just copying code from bmd/election-manager into bsd.

### CODE REVIEW

File: apps/bsd/src/components/ElectionConfiguration.tsx [hunk 1]
```diff
@@ -1,54 +1,291 @@
...
+    try {
+      const files = await window.kiosk!.getFileSystemEntries(
+        path.join(usbPath!, BALLOT_PACKAGE_FOLDER)
+      )
+      setFoundFilenames(
+        files.filter((f) => f.type === 1 && f.name.endsWith('.zip'))
+      )
+      setLoadingFiles(false)
+    } catch (err) {
+      if (err.message.includes('ENOENT')) {
```
Comment (carolinemodic) [inline apps/bsd/src/components/ElectionConfiguration.tsx:91]: This feels gross, but the error is wrapped in another error by kiosk-browser so there is no code on the error I catch here which is normally what you would check for "ENOENT" on

**Review by beausmith — CHANGES_REQUESTED** [review]
I don't think we should use the button from the voting interface here. It's a solution for placing all the election ballot package info in one element, but it seems out of place. 

It's also not clear from the screen cast if you must unselect a configuration before selecting another configuration.

I'd like to suggest the following:

---

# Select Election Configuration

Select one of the following configurations which were automatically found on the USB drive. Optionally, you may select a custom configuration file.

| Export Date | Name | ID | |
|:-|:-|:-|:-|
| Dec 2, 2020 9:30 am | Choctaw County - 2020 General Election | 38ck3c8f | [Select] |
| Dec 2, 2020 9:30 am | Choctaw County - 2020 General Election | 38ck3c8f | [Select] |
| Dec 2, 2020 9:30 am | Choctaw County - 2020 General Election | 38ck3c8f | [Select] |
| Or custom | | | [Select Custom] |

---

**Comment by beausmith** [comment]
Suggestion for Not Configured screen:

Use full-width layout, with margins on the sides. Text all left-aligned.

---

# Load Election Configuration

(USB Drive image here)

You may load an election configuration in the following methods:

- **Insert a USB drive** with election ballot packages exported from Election Manager.
- **Insert an Admin Card** into an attached card reader.
- [Manually Select Configuration File]

---

**Comment by beausmith** [comment]
Suggestion for "No Ballot Packages Found" screen:

---

# No Election Ballot Package Files Found

There were no Election Ballot Package files automatically found on the inserted USB drive. Use Election Manager to export Ballot Package files to this USB drive.

You may also [Manually Select Election Ballot Package]
---

File: apps/bsd/src/components/Button.tsx [hunk 2]
```diff
@@ -22,6 +18,7 @@ interface StyledButtonProps
 
 const buttonStyles = css<StyledButtonProps>`
   border: none;
+  display: inline-block;
```
Comment (carolinemodic) [inline apps/bsd/src/components/Button.tsx:21]: Button already has this by default but label doesn't so I added this for LabelButton's purpose so container elements adjust their size properly

**Comment by carolinemodic** [comment]
@beausmith Back to you, added screenshots in the PR description. The USB image looks a little weird to me still, there is so much blank space in the image that its hard to figure out how it should line up, but I'm also happy to just commit this and then revisit the CSS later (or have you polish it) 

@eventualbuddha would be good to get your eyes on the kiosk browser/reading file interactions here

File: apps/bsd/src/lib/usbstick.ts [hunk 3]
```diff
@@ -55,4 +64,5 @@ export const doUnmount = async (): Promise<void> => {
   }
 
   window.kiosk!.unmountUsbDrive(device.deviceName)
+  return new Promise((resolve) => setTimeout(resolve, 2000))
```
Comment (benadida) [inline apps/bsd/src/lib/usbstick.ts:67]: I would suggest making this longer, say 10s. Until we find a signal that confirms that the data has been written to disk, better safe than sorry, and ejecting disks doesn't happen all that often.

File: apps/bsd/src/util/filenames.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
+import moment from 'moment'
+
+const SECTION_SEPERATOR = '__'
```
Comment (benadida) [inline apps/bsd/src/util/filenames.ts:3]: nit: `SEPARATOR`

File: apps/bsd/src/util/filenames.ts [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+const SUBSECTION_SEPERATOR = '_'
+const WORD_SEPERATOR = '-'
```
Comment (benadida) [inline apps/bsd/src/util/filenames.ts:5]: same nit: `SEPARATOR`

**Review by benadida — APPROVED** [review]
nice, thorough testing. Looks good to me.

File: apps/bsd/src/util/filenames.ts (hunk 4 above)
Comment (carolinemodic) [reply apps/bsd/src/util/filenames.ts:3]: lol whoops

**Pull request merged**